src/loadScript.ts

```diff
@@ -0,0 +1,22 @@
+export const loadScript = (
+	src: string,
+	props?: Omit<Partial<HTMLScriptElement>, 'src' | 'onload' | ' onerror'>,
+): Promise<Event | undefined> =>
+	new Promise((resolve, reject) => {
+		if (document.querySelector(`script[src="${src}"]`)) {
```

Comment:
Do you need to check for `src` being protocol-less or are we happy to just reload if `//` vs `https://`?

Reply:
Also, could we see bugs from trying to load a script with attributes but already loaded without? I don't know the use cases of this, but we might want to log something?

Reply:
hmm, that's a good question. i wonder if we should check for more than just that attribute? e.g. i wonder if you loop through `document.scripts` you get a more fully fledged object?

if we created the script element and only inserted it if no others matched it, it might be more accurate?

Reply:
interesting... 
![image](https://user-images.githubusercontent.com/867233/95100017-9122df00-0728-11eb-8e32-19cd1ac8e290.png)

Reply:
> Also, could we see bugs from trying to load a script with attributes but already loaded without? I don't know the use cases of this, but we might want to log something?

i think if its already in the page, we should be good to return early?

Reply:
> i think if its already in the page, we should be good to return early?

Yeah, i can't think of anything that would modify a script's behaviour based on attributes of the scripts tag, that would break down the line